File: rust/src/models/batch_job_detail_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{
    batch_job_id::BatchJobId, batch_job_status::BatchJobStatus, batch_job_type::BatchJobType,
};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct BatchJobDetailResponse {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub completed_at: Option<String>,

    pub created_at: String,

    pub created_by: String,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub error_csv_url: Option<String>,

    pub failed_items: i32,

    pub failure_count: i32,

    pub has_error_csv: bool,

    pub has_output: bool,

    pub id: BatchJobId,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub input_file_name: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub input_file_url: Option<String>,

    pub job_type: BatchJobType,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub output_url: Option<String>,

    pub processed_items: i32,

    pub status: BatchJobStatus,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub total_items: Option<i32>,
}

impl BatchJobDetailResponse {
    pub fn new(
        created_at: String,
        created_by: String,
        failed_items: i32,
        failure_count: i32,
        has_error_csv: bool,
        has_output: bool,
        id: BatchJobId,
        job_type: BatchJobType,
        processed_items: i32,
        status: BatchJobStatus,
    ) -> Self {
        Self {
            completed_at: None,
            created_at,
            created_by,
            error_csv_url: None,
            failed_items,
            failure_count,
            has_error_csv,
            has_output,
            id,
            input_file_name: None,
            input_file_url: None,
            job_type,
            output_url: None,
            processed_items,
            status,
            total_items: None,
        }
    }
}

File: typescript/src/models/batchJobDetailResponse.ts
// this file is @generated
import { type BatchJobId, BatchJobIdSerializer } from "./batchJobId";
import { type BatchJobStatus, BatchJobStatusSerializer } from "./batchJobStatus";
import { type BatchJobType, BatchJobTypeSerializer } from "./batchJobType";

export interface BatchJobDetailResponse {
  completedAt?: Date | null;

  createdAt: Date;

  createdBy: string;

  errorCsvUrl?: string | null;

  failedItems: number;

  failureCount: number;

  hasErrorCsv: boolean;

  hasOutput: boolean;

  id: BatchJobId;

  inputFileName?: string | null;

  inputFileUrl?: string | null;

  jobType: BatchJobType;

  outputUrl?: string | null;

  processedItems: number;

  status: BatchJobStatus;

  totalItems?: number | null;
}

export const BatchJobDetailResponseSerializer = {
  _fromJsonObject(object: any): BatchJobDetailResponse {
    return {
      completedAt:
        object["completed_at"] != null ? new Date(object["completed_at"]) : undefined,
      createdAt: new Date(object["created_at"]),
      createdBy: object["created_by"],
      errorCsvUrl: object["error_csv_url"],
      failedItems: object["failed_items"],
      failureCount: object["failure_count"],
      hasErrorCsv: object["has_error_csv"],
      hasOutput: object["has_output"],
      id: BatchJobIdSerializer._fromJsonObject(object["id"]),
      inputFileName: object["input_file_name"],
      inputFileUrl: object["input_file_url"],
      jobType: BatchJobTypeSerializer._fromJsonObject(object["job_type"]),
      outputUrl: object["output_url"],
      processedItems: object["processed_items"],
      status: BatchJobStatusSerializer._fromJsonObject(object["status"]),
      totalItems: object["total_items"],
    };
  },

  _toJsonObject(self: BatchJobDetailResponse): any {
    return {
      completed_at: self.completedAt,
      created_at: self.createdAt,
      created_by: self.createdBy,
      error_csv_url: self.errorCsvUrl,
      failed_items: self.failedItems,
      failure_count: self.failureCount,
      has_error_csv: self.hasErrorCsv,
      has_output: self.hasOutput,
      id: BatchJobIdSerializer._toJsonObject(self.id),
      input_file_name: self.inputFileName,
      input_file_url: self.inputFileUrl,
      job_type: BatchJobTypeSerializer._toJsonObject(self.jobType),
      output_url: self.outputUrl,
      processed_items: self.processedItems,
      status: BatchJobStatusSerializer._toJsonObject(self.status),
      total_items: self.totalItems,
    };
  },
};
